src/Relude/Extra/Foldable1.hs

```diff
@@ -1,54 +1,116 @@
+{-# LANGUAGE InstanceSigs #-}
+
 module Relude.Extra.Foldable1
     ( Foldable1 (..)
-    , traverse1_
-    , for1_
     ) where
 
-import Relude.Applicative (Applicative, pass)
-import Relude.Base (Ord)
-import Relude.Foldable.Reexport (Foldable (foldr))
-import Relude.Function (flip, id, (.))
-import Relude.Functor.Reexport (Compose (..), Identity (..))
-import Relude.Monoid (Semigroup, (<>))
+import Relude hiding (Product (..), Sum (..))
+
+import Data.Functor.Product (Product (..))
+import Data.Functor.Sum (Sum (..))
+import qualified Data.Semigroup as SG
 
-import Control.Applicative ((*>))
-import Data.List.NonEmpty (NonEmpty (..))
-import Data.Semigroup (First (..), Last (..), Max (..), Min (..))
+-- | The class of foldable data structures that cannot be empty.
 
 class Foldable t => Foldable1 t where
-    foldMap1   :: Semigroup m => (a -> m) -> t a -> m
+    {- | Map each element of the non-empty structure to a semigroup, and combine the results.
+
+    >>> foldMap1 SG.Sum (1 :| [2, 3, 4])
+    Sum {getSum = 10}
+    -}
+    foldMap1 :: Semigroup m => (a -> m) -> t a -> m
 
+    {- | Combine the elements of a non-empty structure using a semigroup.
+
+    >>> fold1 (4 :| [5, 10 :: SG.Product Int])
+    Product {getProduct = 200}
+    -}
     fold1 :: Semigroup m => t m -> m
     fold1 = foldMap1 id
 
+    {- | Convert a non-empty data structre to a NonEmpty list.
+
+    >>> toNonEmpty (Identity 2)
+    2 :| []
+    -}
     toNonEmpty :: t a -> NonEmpty a
     toNonEmpty = foldMap1 (:|[])
 
+    {- | The first element of a non-empty data structure.
+
+    >>> head1 (1 :| [2, 3, 4])
+    1
+    -}
     head1 :: t a -> a
-    head1 = getFirst . foldMap1 First
+    head1 = SG.getFirst #. foldMap1 SG.First
 
+    {- | The last element of a non-empty data structure.
+
+    >>> last1 (1 :| [2, 3, 4])
+    4
+    -}
     last1 :: t a -> a
-    last1 = getLast . foldMap1 Last
+    last1 = SG.getLast #. foldMap1 SG.Last
 
-    maximum1 :: Ord a => t a -> a
-    maximum1 = getMax . foldMap1 Max
+    {- | The largest element of a non-empty data structure.
 
-    minimum1 :: Ord a => t a -> a
-    minimum1 = getMin . foldMap1 Min
+    >>> maximum1 (32 :| [64, 8, 128, 16])
+    128
+    -}
+    maximum1 :: Ord a => t a -> a
+    maximum1 = SG.getMax #. foldMap1 SG.Max
 
-traverse1_ :: (Foldable1 t, Applicative f) => (a -> f b) -> t a -> f ()
-traverse1_ f = foldr ((*>) . f) pass
+    {- | The smallest elemenet of a non-empty data structure.
 
-for1_ :: (Foldable1 t, Applicative f) => t a -> (a -> f b) -> f ()
-for1_ = flip traverse1_
+    >>> minimum1 (32 :| [64, 8, 128, 16])
+    8
+    -}
+    minimum1 :: Ord a => t a -> a
+    minimum1 = SG.getMin #. foldMap1 SG.Min
 
 instance Foldable1 NonEmpty where
+    foldMap1 :: Semigroup m => (a -> m) -> NonEmpty a -> m
```

Comment:
I really like type signatures in instances, especially for classes like `Foldable`! This will help beginners a lot!